omod/src/main/webapp/pages/runReport.gsp

```diff
@@ -177,6 +192,31 @@ ${ ui.includeFragment("appui", "messages", [ codes: [
                                 initialValue: it.defaultValue ?: sessionContext.sessionLocation,
                                 classes: ["drop-down-list"]
                         ])}
+                    <% } else if (it.type == org.openmrs.EncounterType) { %>
+                        ${ ui.includeFragment("uicommons", "field/encounterType", [
+                                formFieldName: "parameterValues[" + it.name + "]",
+                                label: it.labelOrName,
+                                initialValue: it.defaultValue,
+                                classes: ["drop-down-list"]
+                        ])}
+                    <% } else if (it.type == java.lang.String && it.name.equalsIgnoreCase("Quarter")) { %>
+                        ${ ui.includeFragment("uicommons", "field/dropDown", [
+                                label: it.labelOrName,
+                                formFieldName: "parameterValues[" + it.name + "]",
+                                options: quartersOptions,
+                                classes: ["drop-down-list"],
+                                hideEmptyLabel: true
+                        ])}
+                    <% } else if (it.type == java.lang.String && it.name.equalsIgnoreCase("Locale")) { %>
+                        ${ ui.includeFragment("uicommons", "field/dropDown", [
+                                label: it.labelOrName,
+                                formFieldName: "parameterValues[" + it.name + "]",
+                                options: localeOptions,
+                                classes: ["drop-down-list"],
+                                hideEmptyLabel: true
+                                initialValue: it.defaultValue ?: sessionContext.sessionLocation,
```

Comment:
Initial value of session location for locale?

Reply:
Hi @dkayiwa 
Your are right, that didn't seem  to be correct

I also noticed that since my last commit a ```,``` after ```true``` was missing
It should now be solved

Reply:
@icrc-toliveira which brings me to another question. Did you actually compile and test these changes in the running web application?